docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/jira.md

```diff
@@ -0,0 +1,651 @@
+                        text: " "
+                      - type: text
+                        text: "${{ imageFrom(vars.imageRepo).RepoURL }}:${{ imageFrom(vars.imageRepo).Tag }}"
+                        marks:
+                          - type: code
+                  - type: paragraph
+                    content:
+                      - type: text
+                        text: "Project:"
+                        marks:
+                          - type: strong
+                      - type: text
+                        text: " "
+                      - type: text
+                        text: "${{ ctx.project }}"
+                        marks:
+                          - type: code
+              labels:
+                - "automated-deployment"
+                - "env-${{ ctx.stage.name }}"
+                - "release-${{ imageFrom(vars.imageRepo).Tag }}"
+                - "project-${{ ctx.project }}"
+
+        # Update application
+        - as: update-app
+          uses: argocd-update
+          config:
+            apps:
+              - name: test-app
+                namespace: argocd
+                sources:
+                  - repoURL: https://github.com/company/app-config.git
+                    kustomize:
+                      images:
+                        - repoURL: public.ecr.aws/nginx/nginx
+                          tag: ${{ imageFrom("public.ecr.aws/nginx/nginx").Tag }}
+
+        # Add progress comment
+        - as: comment-on-ticket
+          uses: jira
+          config:
+            credentials:
+              secretName: jira
+            commentOnIssue:
+              issueKey: ${{ outputs['create-deployment-ticket'].key }}
+              body: "Release ${{ imageFrom(vars.imageRepo).Tag }} has been promoted to ${{ ctx.stage.name }} environment at ${{ ctx.promotion.creationTimestamp }}. Freight: ${{ ctx.targetFreight.name }}. Ready for testing."
+
+        # Cleanup on failure
+        - as: on-failure-cleanup-issue
+          uses: jira
+          if: ${{ failure() }}
+          config:
+            credentials:
+              secretName: jira
+            deleteIssue:
+              issueKey: ${{ outputs['create-deployment-ticket'].key }}
+              deleteSubtasks: true
+
+---
+apiVersion: kargo.akuity.io/v1alpha1
+kind: Stage
+metadata:
+  name: stage
+  namespace: kargo-proj
+spec:
+  requestedFreight:
+    - origin:
+        kind: Warehouse
+        name: nginx
+      sources:
+        stages:
+          - test
+  promotionTemplate:
+    spec:
+      vars:
+        - name: imageRepo
+          value: public.ecr.aws/nginx/nginx
+      steps:
+        # Wait for manual approval to proceed to staging
+        - as: wait-approval
+          uses: jira
+          config:
+            credentials:
+              secretName: jira
+            waitForStatus:
+              issueKey: ${{ freightMetadata(ctx.targetFreight.name, 'jira-issue-key') }}
+              expectedStatus: STAGED
```

Comment:
The workflow is odd. We wait for a STAGED status in order to promote _into_ staging? That feels backwards. Same is true for prod further down. We wait for a RELEASED status before promoting into prod. These statuses sound very much like the thing that's being done has _already_ been done. I assume all these statuses are arbitrary or semi-arbitrary? (I haven't used Jira in a long time.) Can we just use different statuses to add clarity? Something like APPROVED_FOR_STAGING?